SIP 4.19.7+ allows to document template based classes

3nids opened the pull request:
it'll be a bit tricky to bring this in the documentation

* dealing with this in sipify is wrong since the source must be independant of the compiling tools (i.e. same source for all)
* sip does not have any preprocessing
* %If directive leads to a syntax error (code gets evaluated even if the condition is not entered)

The only solution I see for the moment is to have an extra option in sipify to output the Docstring where wanted and to rewrite all the SIP files when building the API doc. We will keep the sip files without Docstrings for template based classes in the main QGIS sources. 

so to create these, one needs to run
`./scripts/sipify_all -template-doc`

3nids committed:
SIP 4.19.7+ allows to document template based classes

nirvn commented on python/core/qgsrange.sip:
```diff
@@ -246,9 +246,9 @@ Returns true if this range overlaps another range.
 
 
-typedef QgsTemporalRange< QDate > QgsDateRange;
+typedef QgsTemporalRange< QDate > QgsDateRange ;
```
Unwanted extra space?

3nids replied:
ok, fixed

3nids committed:
[sipify] use configuration file for SIP sources (*.sip.in)

to allow preprocessing the files to handle version specific features such as template based classes documentation in 4.19.7+

3nids committed:
rename *.sip to *.sip.in for the auto generated sip files

3nids commented:
I decided to go for the input configuration files for all auto-generated sip files.
This way, the code is properly hosted on github with all features.

3nids committed:
[sipify] remove space before semi-column

3nids committed:
run sipify

3nids commented:
anybody against the idea of having input files for all sip files in source?

3nids committed:
version compare is not strict

also fix sipify test

3nids committed:
fix variable used for configuration

3nids committed:
fix .gitignore to add missing sip files

nyalldawson commented:
@3nids what's the new approach for updating sip bindings? I noticed that the prepare_commit script no longer works since this change

3nids commented:
should be fixed now, you'll have to rebase.
sorry for that!
